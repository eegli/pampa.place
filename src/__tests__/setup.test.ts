/* 
There are a lot of Google implementations that are used in the thunks
that need to be mocked properly, e.g. StreetViewPreference and
StreetViewSource (see jest-setup.ts). Make sure they have a correctly
defined initial implementation that can later be overwritten
individually for each test. 
*/

describe('__Google mock setup__', () => {
  it('street view service resolves', async () => {
    const service = new google.maps.StreetViewService();
    expect(service).toBeTruthy();
    expect(await service.getPanorama({})).toMatchInlineSnapshot(`
Object {
  "data": Object {
    "copyright": "Copyright",
    "imageDate": "August 2020",
    "links": Array [],
    "location": Object {
      "description": "Fake panorama description",
      "latLng": Object {
        "lat": 0,
        "lng": 0,
      },
      "pano": "69",
      "shortDescription": null,
    },
    "tiles": Object {
      "centerHeading": 0,
      "getTileUrl": [Function],
      "tileSize": Size {
        "height": 1,
        "toString": [MockFunction],
        "width": 1,
      },
      "worldSize": Size {
        "height": 1,
        "toString": [MockFunction],
        "width": 1,
      },
    },
  },
}
`);

    jest
      .spyOn(google.maps.StreetViewService.prototype, 'getPanorama')
      // @ts-ignore
      .mockImplementation(() => {
        return Promise.resolve({ok: true});
      });
    expect(await service.getPanorama({})).toMatchInlineSnapshot(`
Object {
  "ok": true,
}
`);
  });
  it('enums are defined', () => {
    expect(google.maps.StreetViewPreference).toMatchInlineSnapshot(`
Object {
  "BEST": "best",
  "NEAREST": "nearest",
}
`);
    expect(google.maps.StreetViewSource).toMatchInlineSnapshot(`
Object {
  "DEFAULT": "default",
  "OUTDOOR": "outdoor",
}
`);
  });
});

export {};
