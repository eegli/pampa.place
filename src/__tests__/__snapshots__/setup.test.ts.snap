// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test global setup has global definitions 1`] = `
Object {
  "Circle": [Function],
  "Data": [Function],
  "ImageMapType": [MockFunction],
  "LatLng": [Function],
  "LatLngBounds": [Function],
  "MVCArray": [Function],
  "MVCObject": [Function],
  "Map": [Function],
  "MapCanvasProjection": [Function],
  "MapPanes": [Function],
  "MapTypeId": Object {
    "HYBRID": "hybrid",
    "ROADMAP": "roadmap",
    "SATELLITE": "satellite",
    "TERRAIN": "terrain",
  },
  "Marker": [MockFunction],
  "OverlayView": [Function],
  "Point": [Function],
  "Polyline": [Function],
  "Size": [Function],
  "StreetViewPanorama": [Function],
  "StreetViewPreference": Object {
    "BEST": "best",
    "NEAREST": "nearest",
  },
  "StreetViewService": [Function],
  "StreetViewSource": Object {
    "DEFAULT": "default",
    "OUTDOOR": "outdoor",
  },
  "VisibleRegion": [Function],
  "event": Object {
    "addListenerOnce": [MockFunction],
  },
  "places": Object {
    "Autocomplete": [Function],
    "SearchBox": [Function],
  },
}
`;

exports[`Test global setup street view service resolves: resp 1 1`] = `
Object {
  "data": Object {
    "copyright": "Copyright",
    "imageDate": "August 2020",
    "links": Array [],
    "location": Object {
      "description": "Fake panorama description",
      "latLng": Object {
        "lat": 0,
        "lng": 0,
      },
      "pano": "69",
      "shortDescription": null,
    },
    "tiles": Object {
      "centerHeading": 0,
      "getTileUrl": [Function],
      "tileSize": Size {
        "height": 1,
        "toString": [MockFunction],
        "width": 1,
      },
      "worldSize": Size {
        "height": 1,
        "toString": [MockFunction],
        "width": 1,
      },
    },
  },
}
`;

exports[`Test global setup street view service resolves: resp 2 1`] = `
Object {
  "ok": true,
}
`;
